iceoryx_posh/test/moduletests/test_roudi_port_introspection.cpp

```diff
@@ -123,190 +138,163 @@ class PortIntrospection_test : public Test
-    SenderPort_MOCK portThroughput_mock2;
-    SenderPort_MOCK portreceiverPortData_mock2;
-    EXPECT_THAT(introspection->registerSenderPort(std::move(senderPortImpl_mock2),
-                                                  std::move(portThroughput_mock2),
-                                                  std::move(portreceiverPortData_mock2)),
+    iox::popo::PublisherPortData m_publisherPortDataPortGeneric2{m_serviceDescription, "Foo", &m_memoryManager};
+    iox::popo::PublisherPortData m_publisherPortDataThroughput2{m_serviceDescription, "Foo", &m_memoryManager};
+    iox::popo::PublisherPortData m_publisherPortDataSubscriberData2{m_serviceDescription, "Foo", &m_memoryManager};
+
+    EXPECT_THAT(introspection->registerPublisherPort(&m_publisherPortDataPortGeneric2,
+                                                     &m_publisherPortDataThroughput2,
+                                                     &m_publisherPortDataSubscriberData2),
                 Eq(false));
-    EXPECT_THAT(mockSender->enableDoDeliverOnSubscription, Eq(1));
-    EXPECT_THAT(mockPort->enableDoDeliverOnSubscription, Eq(1));
 }
 
+
 TEST_F(PortIntrospection_test, sendPortData_EmptyList)
 {
-    using Topic = iox::roudi::PortIntrospectionFieldTopic;
-
-    auto chunk = std::unique_ptr<ChunkMock<Topic>>(new ChunkMock<Topic>);
-
-    m_senderPortImpl_mock->reserveSampleReturn = chunk->chunkHeader();
+    EXPECT_CALL(m_introspectionAccess.getPublisherPort().value(), sendChunk(_)).Times(1);
 
     m_introspectionAccess.sendPortData();
 
-    // topic contains no sender or receiver ports but 0xFF bytes are overwritten
-
-    EXPECT_THAT(m_senderPortImpl_mock->deliverChunk, Eq(1));
-    EXPECT_THAT(chunk->sample()->m_senderList.size(), Eq(0));
-    EXPECT_THAT(chunk->sample()->m_receiverList.size(), Eq(0));
+    EXPECT_THAT(m_introspectionAccess.getPublisherPort().value().m_chunk.sample()->m_publisherList.size(), Eq(0));
+    EXPECT_THAT(m_introspectionAccess.getPublisherPort().value().m_chunk.sample()->m_subscriberList.size(), Eq(0));
 }
 
+
 TEST_F(PortIntrospection_test, sendThroughputData_EmptyList)
 {
-    using Topic = iox::roudi::PortThroughputIntrospectionFieldTopic;
-
-    auto chunk = std::unique_ptr<ChunkMock<Topic>>(new ChunkMock<Topic>);
-
-    m_portThroughput_mock->reserveSampleReturn = chunk->chunkHeader();
+    /// @todo #252 re-add port throughput for v1.0?
+    // using Topic = iox::roudi::PortThroughputIntrospectionFieldTopic;
 
-    m_introspectionAccess.sendThroughputData();
+    // auto chunk = std::unique_ptr<ChunkMock<Topic>>(new ChunkMock<Topic>);
 
-    // topic contains no sender or receiver ports but 0xFF bytes are overwritten
+    // EXPECT_CALL(m_introspectionAccess.getPublisherPortThroughput().value(), tryAllocateChunk(_))
+    //     .WillOnce(Return(ByMove(iox::cxx::success<iox::mepoo::ChunkHeader*>(chunk->chunkHeader()))));
+    // EXPECT_CALL(m_introspectionAccess.getPublisherPortThroughput().value(), sendChunk(_)).Times(1);
```

Comment:
Please replace out commented lines with todo and commit hash

Reply:
how do you know the commit hash before the commit 😉

Reply:
Done.